L1Trigger/TrackTrigger/src/HitPatternHelper.cc

```diff
@@ -0,0 +1,273 @@
+//
+//  Created by J.Li on 1/23/21.
+//
+
+#include "L1Trigger/TrackTrigger/interface/HitPatternHelper.h"
+#include "L1Trigger/TrackFindingTMTT/interface/KFbase.h"
+#include "L1Trigger/TrackFindingTMTT/interface/TrackerModule.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <algorithm>
+#include <cmath>
+
+namespace hph {
+
+  Setup::Setup(const edm::ParameterSet& iConfig, const tt::Setup& setupTT)
+      : iConfig_(iConfig),
+        setupTT_(setupTT),
+        layerIds_(),
+        layermap_(),
+        nEtaRegions_(tmtt::KFbase::nEta_ / 2),
+        nKalmanLayers_(tmtt::KFbase::nKFlayer_) {
+    for (const tt::SensorModule& sm : setupTT_.sensorModules()) {
+      layerIds_.push_back(std::make_pair(sm.layerId(), sm.barrel()));
+    }
+    sort(layerIds_.begin(), layerIds_.end(), smallerID);
+    layerIds_.erase(unique(layerIds_.begin(), layerIds_.end(), equalID), layerIds_.end());  //Keep only unique layerIds
+    // Converting tmtt::KFbase::layerMap_ to a format that is acceptatble by HitPatternHelper
+    for (int i = 0; i < nEtaRegions_; i++) {
+      for (int j = 0; j < (int)layerIds_.size(); j++) {
+        int layer = nKalmanLayers_;
```

Comment:
```suggestion
        int layer;
```
as it is overwritten anyhow

Reply:
Done. @Jingyan95 please note this.